langchain4j-azure-cosmos/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/mongo/vcore/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,622 @@
+                    MongoClientSettings.builder()
+                            .applyConnectionString(new ConnectionString(connectionString))
+                            .applicationName(applicationName)
+                            .build());
+        }
+
+        MongoDatabase database = mongoClient.getDatabase(databaseName);
+        // create collection if not exist
+        if (!isCollectionExist(database, collectionName)) {
+            createCollection(database, collectionName, getOrDefault(createCollectionOptions, new CreateCollectionOptions()));
+        }
+        this.collection = database.getCollection(collectionName, AzureCosmosDbMongoVCoreDocument.class).withCodecRegistry(codecRegistry);
+
+        // create index if not exist
+        if (Boolean.TRUE.equals(createIndex) && !isIndexExist(this.indexName)) {
+            createIndex(this.indexName, collectionName, database);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String add(Embedding embedding) {
+        String id = randomUUID();
+        add(id, embedding);
+        return id;
+    }
+
+    @Override
+    public void add(String id, Embedding embedding) {
+        addInternal(id, embedding, null);
+    }
+
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+//        List<Double> queryVector = referenceEmbedding.vectorAsList().stream()
+//                .map(Float::doubleValue)
+//                .collect(toList());
+
+        List<Bson> pipeline = new ArrayList<>();
+
+        switch (this.kind) {
+            case VECTOR_IVF:
+                pipeline = getPipelineDefinitionVectorIVF(referenceEmbedding, maxResults);
+                break;
+            case VECTOR_HNSW:
+                pipeline = getPipelineDefinitionVectorHNSW(referenceEmbedding, maxResults);
+                break;
+        }
+
+        try {
+            AggregateIterable<BsonDocument> results = collection.aggregate(pipeline, BsonDocument.class);
+            List<BsonDocument> abc = StreamSupport.stream(results.spliterator(), false).
```

Comment:
what is abc? please remove if not needed

Reply:
Removed the code.